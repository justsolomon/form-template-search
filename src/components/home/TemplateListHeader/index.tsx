import { TemplateCategory } from 'types/global';
import styles from './templateListHeader.module.scss';

interface TemplateListHeaderProps {
  category: 'All' | TemplateCategory;
  numOfTemplates: number;
}

const TemplateListHeader = ({
  category,
  numOfTemplates,
}: TemplateListHeaderProps) => {
  return (
    <div className={styles['list-header']}>
      <h1
        className={styles['list-header__title']}
      >{`${category} Templates`}</h1>

      <p
        className={styles['list-header__templates-number']}
      >{`${numOfTemplates} templates`}</p>
    </div>
  );
};

export default TemplateListHeader;
