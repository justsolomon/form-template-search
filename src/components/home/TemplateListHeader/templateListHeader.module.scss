@import 'styles/variables';
@import 'styles/mixins';

.list-header {
  @include flex-center();

  margin-bottom: 24px;
  justify-content: space-between;

  &__title {
    font-size: $font-size--md;
    line-height: $font-size--lg;
  }

  &__templates-number {
    color: $secondary-color;
  }
}
