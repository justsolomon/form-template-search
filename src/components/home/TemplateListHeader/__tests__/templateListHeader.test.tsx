import { render } from '@testing-library/react';
import TemplateListHeader from '..';

describe('<TemplateListHeader />', () => {
  it('should render the TemplateListHeader component correctly', () => {
    const { asFragment } = render(
      <TemplateListHeader category="All" numOfTemplates={5} />,
    );

    expect(asFragment).toMatchSnapshot();
  });
});
