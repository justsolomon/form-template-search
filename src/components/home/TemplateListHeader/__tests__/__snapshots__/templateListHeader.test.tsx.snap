// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TemplateListHeader /> should render the TemplateListHeader component correctly 1`] = `[Function]`;
